specs/lifespan.rst

```diff
@@ -0,0 +1,55 @@
+=================
+Lifespan Protocol
+=================
+
+**Version**: 2.1 (2018-07-01)
+
+The Lifespan ASGI sub-specification outlines how to communicate
+lifespan events such as startup and cleanup within ASGI. The lifespan
+being refered to is that of main event loop.
+
+
+Scope
+'''''
+
+The lifespan scope exists for the duration of the event loop. The
+scope itself contains basic metadata,
+
+* ``type``: ``lifespan``
+
+* ``asgi``: A dictionary containing a ``version`` key, which itself is
+  a unicode string version identifier, defaults to ``2.0``.
+
+
+Startup
+'''''''
+
+Sent when the server is ready to startup and recieve connections, but
```

Comment:
What would this mean if the server didn't have incoming connections, but instead called out to an external API (like some of the chatbot servers I've whipped up)? Is there a better way to phrase "connections"?

Reply:
I think we should also change the asgi file as well, I was matching 

> This first callable is called whenever a new connection comes in to the protocol server, and creates a new instance of the application per connection (the instance is the object that this first callable returns).

I'm not sure how else to word it though, maybe just in terms of application instances?